operator/v1/types_dns.go

```diff
@@ -170,6 +184,35 @@ type ForwardPlugin struct {
 	// +optional
 	// +kubebuilder:default:="Random"
 	Policy ForwardingPolicy `json:"policy,omitempty"`
+
+	// transport allows cluster administrators to opt-in to using a DNS-over-TLS
+	// connection between cluster DNS and an upstream resolver(s). Configuring
+	// TLS as the transport at this level without configuring a CABundle will
+	// result in the system certificates being used to verify the serving
+	// certificate of the upstream resolver(s).
+	//
+	// +optional
+	Transport DNSTransport `json:"transport,omitempty"`
+
+	// caBundle references a ConfigMap that must contain either a single
+	// CA Certificate or a CA Bundle (in the case of multiple upstreams signed
+	// by different CAs). This allows cluster administrators to provide their
+	// own CA or CA bundle for validating the certificate of upstream resolvers.
+	//
+	// 1. The configmap must contain a `cabundle.crt` key
+	// 2. The value must be a PEM encoded certificate or bundle.
+	// 3. The administrator must create this configmap in the openshift-config namespace.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	CABundle v1.ConfigMapNameReference `json:"caBundle,omitempty"`
+
+	// serverName is the upstream server to connect to for DNS resolution.
```

Comment:
Why do we set one server for 15 upstreams again?

Reply:
If I understand your question correctly, you're asking why we're limited to forwarding a zone to multiple upstreams with one servername? If that understanding is correct, it's because of a limitation in the forward plugin in CoreDNS: `Multiple upstreams are still allowed <snip> but they have to use the same tls_servername`.

I believe the context for this limitation is that in the Corefile you define multiple IPs as the upstreams but they must all be in the SAN of the TLS cert which has a CN matching the `tls_servername` value.

Reply:
> t's because of a limitation in the forward plugin in CoreDNS: Multiple upstreams are still allowed <snip> but they have to use the same tls_servername.

why don't we fix coredns?

Reply:
I'll defer to @Miciah on addressing this long term as that will probably fall to the NetEdge team. I think from an API perspecitve we could do both, right? Right now we can define one `tls_servername` for a set of upstreams and then pending future upstream work to support multiple `forward` blocks we could add the configuration options to do that.